File: 2619.java
/* @author: */
import java.util.Scanner;
import java.util.Arrays;
public class Main{

 public static void main(String args[])
{
 Scanner sc=new Scanner(System.in);

	int a,b;
       a=sc.nextInt();
       b=sc.nextInt();
	int p1,p2,p3,d1,d2,d3;
	p1=(int)Math.sqrt(a*1.0);
	if(p1*p1< a) p1++;
	d1=(int)Math.sqrt(b*1.0);
	if(d1*d1< b) d1++;
	int v1=(p1-1)*(p1-1);
	int v2=(d1-1)*(d1-1);
	p2=(a-v1+1)/2;
	d2=(b-v2+1)/2;
	p3=(p1*p1-a)/2+1;
	d3=(d1*d1-b)/2+1;
	int ans=Math.abs(p1-d1)+Math.abs(p2-d2)+Math.abs(p3-d3);
	System.out.printf("%d\n",ans);
 }
}

File: 2619.cc
#include <stdio.h>
#include <stdlib.h>
#include <math.h>
int main() {
  int temp, n, m, s, flag = 3, cn, cm, zb, yb, cc, k, sj;
  while (scanf("%d%d", &m, &n) == 2) {
    flag = 3;
    s = 0;
    k = 0;
    if (m > n) {
      temp = m;
      m = n;
      n = temp;
    }
    cn = ceil(sqrt(n));
    cm = ceil(sqrt(m));
    cc = abs(cn - cm);
    zb = m + (cn - 1) * (cn - 1) - (cm - 1) * (cm - 1);
    yb = zb + 2 * cc;

    if (n >= zb && n <= yb) {
      s = 2 * (cc);
      k = 1;
    } else {
      if (n < zb)
        s = 2 * (cc) + abs(n - zb);
      else
        s = 2 * (cc) + abs(n - yb);
    }
    sj = m - (cm - 1) * (cm - 1);
    if (abs(n - m) % 2 != (cc) % 2) {
      if (sj % 2 == 1)
        flag = 1;
      else
        flag = 0;
    }
    if (flag == 1 && k == 1) s = s - 1;
    if (flag == 0 && k == 1) s = s + 1;
    printf("%d\n", s);
  }
  return 0;
}
